### PR TITLE
Add a "validate bag" activity

### LINKED ISSUE #19: Feature: Add a BagIt bag validation activity
To address https://github.com/artefactual-sdps/enduro/issues/805 we need an activity to validate a BagIt bag in Enduro. We will likely need to validate bags in other contexts (e.g. https://github.com/artefactual-sdps/preprocessing-base) as well.

**Describe the solution you'd like**

Add a `bagit.ValidateBagActivity` to validate BagIt bags using the https://github.com/LibraryOfCongress/bagit-python library.

**Describe alternatives you've considered**

1. Validating bags with the https://github.com/nyudlts/go-bagit.  We are not confident in the go-bagit library's ability to handle validation edge cases.  We believe the older and more tested https://github.com/LibraryOfCongress/bagit-python will provide better validation results right now.
2. Create our own Go bag validation library.  Again, we are not confident that a home built bag validator would handle validation edge cases, so a mature library like https://github.com/LibraryOfCongress/bagit-python is preferable.

**Additional context**

1. Issue #11 has more information about our decision to use https://github.com/LibraryOfCongress/bagit-python for bag validation
2. As noted in #11 we may use https://github.com/artefactual-labs/bagit-gython to embed the python interpreter, dependencies and bagit-python package in the go binary to avoid additional deployment dependencies

### PR DESCRIPTION
**Pull request by djjuhasz**
Fixes #19.

The expectation is https://github.com/artefactual-labs/bagit-gython will be used as the bag validator. The bag validator is being injected into the activity so the setting up the embedded Python interpreter (which is expensive) is only done once in a workflow even if validate_activity is run multiple times.

### CODE REVIEW

File: bagit/validate_activity.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	logger.V(1).Info("Executing BagIt Validate Activity", "Path", params.Path)
+
+	if err := a.validator.Validate(params.Path); err != nil {
+		return &ValidateActivityResult{
+			Valid: false,
+			Error: err.Error(),
+		}, nil
+	}
+
+	return &ValidateActivityResult{Valid: true}, nil
```
Comment (jraddaoui) [inline bagit/validate_activity.go:55]: I see https://github.com/artefactual-labs/bagit-gython doesn't make a distinction between validation and other type of errors, so I guess we could do the same and just return an empty result and the error from `Validate`. I'd prefer to look for the error directly when calling the activity, which will still be needed, than looking for that error inside the result.

**Review by jraddaoui — APPROVED** [review]
Nice! 

I think returning the error directly would be better, but feel free to ignore.

File: bagit/validate_activity.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	// Path is the full path of the Bag to be validated.
+	Path string
+}
+
+type ValidateActivityResult struct {
+	// Valid is true if the Bag is valid.
+	Valid bool
+
+	// Error is a message indicating why validation failed, and will always be
+	// empty when IsValid is true.
```
[the remark this answers is not in the document]
Comment (djjuhasz) [reply bagit/validate_activity.go:36]: Oops. Thanks. :)

File: bagit/validate_activity.go (hunk 1 above)
Comment (djjuhasz) [reply bagit/validate_activity.go:55]: Yes, when I was looking at Go validation libraries it seemed like it was pretty common to return a validation error as a Go `error`, so I think it's consistent for https://github.com/artefactual-labs/bagit-gython to do the same.  I do think it's problematic that bagit-gython doesn't return a specialized error type to allow differentiating validation errors from application errors though — I think I'll file an issue ticket about that.

I'm inclined to keep the validation "failures" in the `ValidateActivityResult` for a few reasons though:

1. A validation error is NOT an application error, it's an expected outcome of trying to validate an invalid bag.  If validation fails it shouldn't trigger activity retries, and I don't think it should be logged as an activity error.   I know we can set the activity retry policy to not retry on a validation error, but I think it's better not to have to rely on a caller remembering to do that.
2. While we can use Go error type checking to differentiate validation errors from application errors, I'd rather not. Go error type checking is terrible in my opinion — it's one of my least favourite parts of the language. :(  I really want `errors.Is()` to just *work* with dynamic errors, but it just doesn't.  And I think doing type checking with `errors.As()` is ugly and counter-intuitive, e.g. 
  ```go
  var e bagit.ErrValidation
  if (errors.As(e, err)) { return "blah!" }
  ```

**Pull request merged**